posix/unix.c

```diff
@@ -653,9 +687,16 @@ static ssize_t recv(unsigned socket, void *buf, size_t len, int flags, struct so
 		for (;;) {
 			proc_lockSet(&s->lock);
 			if (s->type == SOCK_STREAM) {
+				if (flags & MSG_PEEK) {
+					/* TODO: peek control data */
+					err = _cbuffer_peek(&s->buffer, buf, len);
+					proc_lockClear(&s->lock);
+					break;
```

Comment:
You shouldn't break here. The code from lines 669-728 also applies to the peek `recv()`. Blocking peek `recv()` should be supported. Additionally, note that the non-blocking peek `recv()` currently returns 0 instead of `EAGAIN` when cbuffer is empty. Could you also add support for `MSG_DONTWAIT`? It should be fairly straightforward to implement once `recv()` is fixed.

Reply:
Corrected. If by `MSG_DONTWAIT` support you mean proper behavior when `MSG_DONTWAIT`, `MSG_PEEK` are both set, then it's also done